arithmetization/src/main/java/net/consensys/linea/plugins/readiness/TracerReadinessPlugin.java

```diff
@@ -0,0 +1,140 @@
+
+  /**
+   * Register the needed Besu services.
+   *
+   * @param context the BesuContext to be used.
+   */
+  @Override
+  public void register(final BesuContext context) {
+    super.register(context);
+    this.besuContext = context;
+  }
+
+  @Override
+  protected Map<String, LineaOptionsPluginConfiguration> getLineaPluginConfigMap() {
+    final TracerReadinessCliOptions tracerReadinessCliOptions = TracerReadinessCliOptions.create();
+
+    return Map.of(TracerReadinessCliOptions.CONFIG_KEY, tracerReadinessCliOptions.asPluginConfig());
+  }
+
+  @Override
+  public void start() {
+    // Initialize Vertx
+    final Vertx vertx = Vertx.vertx();
+
+    final Router router = Router.router(vertx);
+
+    // Register the readiness check handler
+    router
+        .get(TRACER_READINESS_ENDPOINT_NAME)
+        .handler(
+            routingContext -> {
+              if (isTracerReady()) {
+                routingContext.response().setStatusCode(200).end(statusResponse("UP"));
+              } else {
+                routingContext.response().setStatusCode(503).end(statusResponse("DOWN"));
+              }
+            });
+
+    TracerReadinessConfiguration configuration =
+        (TracerReadinessConfiguration)
+            getConfigurationByKey(TracerReadinessCliOptions.CONFIG_KEY).optionsConfig();
+
+    // Start the Vertx HTTP server
+    server =
+        vertx
+            .createHttpServer(httpServerOptions(configuration))
+            .requestHandler(router)
+            .listen(
+                configuration.serverPort(),
+                result -> {
+                  final String pluginName = getClass().getSimpleName();
+                  final int port = configuration.serverPort();
+
+                  if (result.succeeded()) {
+                    log.info("[{}] Started listening on port {}", pluginName, port);
+                  } else {
+                    log.error(
+                        "[{}] Failed to start listening on port {}: {}",
+                        pluginName,
+                        port,
+                        result.cause().getMessage());
+                  }
+                });
+  }
+
+  private String statusResponse(final String status) {
+    return new JsonObject(singletonMap("status", status)).encodePrettily();
+  }
+
+  private HttpServerOptions httpServerOptions(final TracerReadinessConfiguration config) {
+    return new HttpServerOptions().setHost(config.serverHost()).setPort(config.serverPort());
+  }
+
+  private boolean isTracerReady() {
+    this.synchronizationService =
+        Optional.ofNullable(synchronizationService)
+            .orElse(BesuServiceProvider.getSynchronizationService(besuContext));
+
+    final RequestLimiter requestLimiter =
+        RequestLimiterDispatcher.getLimiter(
+            RequestLimiterDispatcher.SINGLE_INSTANCE_REQUEST_LIMITER_KEY);
+
+    return !requestLimiter.isNodeAtMaxCapacity() && synchronizationService.isInitialSyncPhaseDone();
```

Comment:
just as a side note to not forget `isInitialSynchPhaseDone` will not give what you are looking after but there's no other option right now.
I open up the `isInSynch` check that if the head of the chain is out of synch here: https://github.com/hyperledger/besu/pull/7665